connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```
@@ -421,19 +421,24 @@ private void removeConnectorTasks(String connName) {
     }
 
     private void updateConnectorTasks(String connName) {
-        if (!worker.isRunning(connName)) {
-            log.info("Skipping update of connector {} since it is not running", connName);
-            return;
-        }
+        try {
+            if (!worker.isRunning(connName)) {
+                log.info("Skipping update of connector {} since it is not running", connName);
+                return;
+            }
 
-        List<Map<String, String>> newTaskConfigs = recomputeTaskConfigs(connName);
-        List<Map<String, String>> oldTaskConfigs = configState.allTaskConfigs(connName);
+            List<Map<String, String>> newTaskConfigs = recomputeTaskConfigs(connName);
+            List<Map<String, String>> oldTaskConfigs = configState.allTaskConfigs(connName);
 
-        if (!newTaskConfigs.equals(oldTaskConfigs)) {
-            removeConnectorTasks(connName);
-            List<Map<String, String>> rawTaskConfigs = reverseTransform(connName, configState, newTaskConfigs);
-            configBackingStore.putTaskConfigs(connName, rawTaskConfigs);
-            createConnectorTasks(connName);
+            if (!newTaskConfigs.equals(oldTaskConfigs)) {
+                removeConnectorTasks(connName);
+                List<Map<String, String>> rawTaskConfigs = reverseTransform(connName, configState, newTaskConfigs);
+                configBackingStore.putTaskConfigs(connName, rawTaskConfigs);
+                createConnectorTasks(connName);
+            }
+        } catch (Throwable t) {
+            // TODO: when this throws errors where do they go
+            log.error("Unable to update connector tasks", t);
```

Left in accidentally?

---

yeah something i need to follow up on but doesn't belong in this PR.
I had some bad mocks which caused a 1000s wait while this method swallowed the errors.